client/src/components/nav_comp/NavBar.js

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+// import { Route } from 'react-router-dom' ;
+import { useSelector } from 'react-redux';
+import LeftNav from './LeftNav'
+import MiddleNav from './MiddleNav'
+import RightNav from './RightNav'
+import RightNavLogged from './RightNavLogged'
+import '../../style/navbar.css'
+
+function NavEditor(){
+    const currentUserId = useSelector(state => state.auth.id);
+    let RightNavSection = RightNav
+    if(currentUserId){
+        RightNavSection = RightNavLogged
+    }
+    return RightNavSection
+}
+
+function NavBar(){
+    const RightSection = NavEditor();
+    return (
+        <div className="outer-nav">
+            <div className="navbar">
+                <LeftNav />
+                <MiddleNav />
+                <RightSection />
+            </div>
+        </div>
+    )
+}
```

Comment:
You can shorten up your code by using a ternary operation to conditionally render the `RightNavLogged` or `RightNav` component based on whether the `currentUserId` is truthy: `{currentUserId ? <RightNavLogged /> : <RightNav />}`.

```suggestion
function NavBar(){
    const currentUserId = useSelector(state => state.auth.id);
    return (
        <div className="outer-nav">
            <div className="navbar">
                <LeftNav />
                <MiddleNav />
                {currentUserId ? <RightNavLogged /> : <RightNav />}
            </div>
        </div>
    )
}
```